test/common/storagelib.py

```diff
@@ -19,18 +19,31 @@
 import os.path
 import re
 import textwrap
+from typing import TYPE_CHECKING, Protocol
 
 from testlib import Error, MachineCase, wait
 
+if TYPE_CHECKING:
```

Comment:
What would the perfect approach be here?

I guess testlib should define MachineProtocol and we just use it here?  And testlib would unconditionally define the fuill version of MachineProtocol and MachineCase derives from it?

Could we do that?

Reply:
Ah good point, btw got this from mypy [mixin classes docs](https://mypy.readthedocs.io/en/stable/more_types.html#mixin-classes)